### DISCUSSION TITLE
Refactor Property Listing Page to Use Component-Based Architecture

### DISCUSSION THREAD

**Discussion by Galope-Jhean** [PUP-BSIT/project-vrzn_appdev#17] [General]
The current implementation of the property listing page is using a monolithic structure. To improve maintainability, readability, and reusability, it is proposed to refactor the page to use a component-based architecture. This will involve breaking down the page into smaller, reusable components.

[SpaceShare.UI/src/app/property/property.component.html](https://github.com/PUP-BSIT/project-vrzn_appdev/blob/08fdc41ad4cbad3056c1be7ea225023b0b999a65/SpaceShare.UI/src/app/property/property.component.html)

Proposed Refactor:

The page will be broken down into the following components:

- Carousel Component - To display the image carousel.
- PropertyDetails Component - To display property details (title, address, price, description, wishlist button).
- OfferForm Component - For the offer submission form.
- PropertyFeatures Component - To display features like bedrooms, area, and guest capacity.
- HostInfo Component - To display the host's information.

**Comment by Dimasacat-Andronicus** [comment]
Your issue has been received and duly noted.

**Comment by Naoe-Adrian** [comment]
this is noted I will try to research about monolithic structure its kinda overwhelming right now but thank you guys for some advices.

**Comment by go-christianharrel** [comment]
Converting this to a discussion to facilitate ongoing and future conversations regarding **Component-Based Architecture**, even after the issue is closed.

**Reply by go-christianharrel** [reply]
Natanggal pala siya as issue when converted to discussion. You may create an issue parin naman from this discussion if ever. Then assign it to someone nalang ule @Galope-Jhean. Thank you!

**Comment by Odrunia-Mark-Ryan** [comment]
Understood, I'll make those changes asap, Thank you!
